moved/src/state_actor.rs

```diff
@@ -449,11 +456,10 @@ impl<
             tx_index += 1;
         }
 
-        let receipts_root = receipts
-            .iter()
-            .map(|v| alloy::rlp::encode(&v.receipt))
-            .map(keccak256)
-            .merkle_root();
+        let receipts_root =
+            alloy_trie::root::ordered_trie_root_with_encoder(&receipts, |rx, buf| {
```

Comment:
Can we explain what this does in a short comment? It seems important.

Reply:
Done in https://github.com/MovedNetwork/op-move/pull/168/commits/d8a95d91410e3c960a60dbdead731e42dcc99ff7